## 4.6 Merging and Exporting data

### This script contains the following points:

#### 01. Import libraries

#### 02. Import datasets & Create a path

#### 03. Dimensions check to understand the basic info of the df

#### 04. Merging dataframes

#### 05. Check if it was a full match between 2 df

#### 06. Export data in pickle format

#### 01. Import libraries

In [40]:
# Import libraries
import pandas as pd
import numpy as np
import os

#### 02. Import datasets & Create a path

In [42]:
# path
path = r'/Users/renatabatista/Other Docs/Germany/CareerFoundry/Data Imersion/4. Python/14-04-2025 Instacart Basket Analysis'

In [43]:
# Import datasets orders_products_prior.csv
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

In [44]:
# Import dataset ords_checked.csv
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'ords_checked.csv'), index_col = False)

#### 03. Dimensions check to understand the basic info of the df

In [46]:
# Check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [47]:
df_ords.head()

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0           0   2539329        1             1                   2   
1           1   2398795        1             2                   3   
2           2    473747        1             3                   3   
3           3   2254736        1             4                   4   
4           4    431534        1             5                   4   

   order_hour_of_day  days_since_prior_order  first_order  
0                  8                     NaN         True  
1                  7                    15.0        False  
2                 12                    21.0        False  
3                  7                    29.0        False  
4                 15                    28.0        False

In [70]:
# Dropping unnecessary column Unnamed: 0 
df_ords = df_ords.drop(['Unnamed: 0'], axis=1)

KeyError: "['Unnamed: 0'] not found in axis"

In [58]:
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  first_order  
0                     NaN         True  
1                    15.0        False  
2                    21.0        False  
3                    29.0        False  
4                    28.0        False

In [62]:
df_ords.shape

(3421083, 7)

In [72]:
df_ords_prior.shape

(32434489, 4)

#### 04. Merging dataframes

In [75]:
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [77]:
df_merged_large.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  first_order  product_id  add_to_cart_order  \
0                     NaN         True         196                  1   
1                     NaN         True       14084                  2   
2                     NaN         True       12427                  3   
3                     NaN         True       26088                  4   
4                     NaN         True       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

#### 05. Check if it was a full match between 2 df

In [80]:
df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

##### The final df has 32,434,489 rows, and each of those rows have information found in both input data sets, as we used an INNER JOIN for the purpose of this project.

#### 06. Export data in pickle format

In [84]:
# Export data to pkl
df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.pkl'))

In [86]:
# Export modified df_ords dataframe (column unnamed deleted)
df_ords.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_final.csv'))